### PR TITLE
refactor: Make com.google.android.material dependency optional

### PR DESCRIPTION
**Pull request by mykola-mokhnach**
https://github.com/appium/appium-espresso-driver/pull/610#issuecomment-701192125

### CODE REVIEW

**Comment by KazuCocoa** [comment]
fyi: not the same, but I'm just trying https://github.com/appium/appium-espresso-driver/pull/615

File: lib/server-builder.js [hunk 1]
```diff
@@ -115,32 +110,45 @@ class ServerBuilder {
...
-      // This ensures that a patched build.gradle will have correct new lines and indentation
-      new RegExp(`(\\s*^\\s*)${_.escapeRegExp(GRADLE_DEPENDENCIES_PLACEHOLDER)}\\s*$`, 'gm'),
-      `$1${dependencyLines}`
-    );
+    const buildPath = path.resolve(this.serverPath, 'app', 'build.gradle');
+    let configuration = await fs.readFile(buildPath, 'utf8');
+    for (const propName of ['additionalAppDependencies', 'additionalAndroidTestDependencies']) {
+      const prefix = propName === 'additionalAppDependencies'
+        ? 'implementation'
+        : 'androidTestImplementation';
```
Comment (KazuCocoa) [inline lib/server-builder.js:140]: 👍 I wondered the same thing (`implementation` and `androidTestImplementation`)

**Review by KazuCocoa — APPROVED** [review]
yey, great

File: lib/server-builder.js [hunk 2]
```diff
@@ -115,32 +110,45 @@ class ServerBuilder {
...
 
-  async insertAdditionalAppDependencies (additionalAppDependencies) {
-    const buildPath = path.resolve(this.serverPath, 'app', 'build.gradle');
-    const originalConfiguration = await fs.readFile(buildPath, 'utf8');
-    const newConfiguration = this.updateDependencyLines(originalConfiguration, additionalAppDependencies);
-    await fs.writeFile(buildPath, newConfiguration, 'utf8');
-  }
+  async insertAdditionalDependencies () {
+    let hasAdditionalDeps = false;
+    for (const propName of ['additionalAppDependencies', 'additionalAndroidTestDependencies']) {
```
[the remark this answers is not in the document]
Comment (mykola-mokhnach) [reply lib/server-builder.js:115]: yep, will change

**Pull request merged**